include/support.h

```diff
@@ -50,6 +50,12 @@
 #define strncasecmp(a, b, n) _strnicmp(a, b, n)
 #endif
 
+#ifdef PAGESIZE
+constexpr size_t sys_pagesize = PAGESIZE;
+#else
+constexpr size_t sys_pagesize = 4096;
+#endif
+
```

Comment:
I'm getting worried about proliferation of constants.

We've got:
 - Meson defining `PAGESIZE` in `config.h`
 - include/mem.h: `#define MEM_PAGESIZE 4096`
 - src/cpu/core_dyn_x86.cpp: `#define PAGESIZE 4096`
 - src/cpu/core_dynrec.cpp: `#define PAGESIZE 4096`
 - src/cpu/dyn_cache.h: `#define PAGESIZE_TEMP PAGESIZE`
 - src/cpu/dyn_cache.h: `#define PAGESIZE_TEMP 4096`
 - src/hardware/memory.cpp: `#define PAGESIZE 4096`
 - src/hardware/mixer.cpp: `#define PAGESIZE 4096`

And now we'll have `sys_pagesize` as another constant.

I think we only want: 
 - Host page size
 - DOS page size (hardcoded to 4096)

Ideally these would be set once, and everyone else would use those.
Would be sweet to get that level of cleanup in this PR :grin:

Reply:
I would like to replace uses of PAGESIZE (for the host pagesize) with `sys_pagesize` (`host_pagesize`?). We need the system pagesize defined at compile time, and the current Meson mechanism is already there. The constexpr gets the rest of the system away from using the `#define`, and than can be used throughout when host pagesize is needed.

`MEM_PAGESIZE` could go to `dos_pagesize`

Reply:
Looks good, and like that approach of having the two constexpr variables!